google-cloud-datastore/src/test/java/com/google/cloud/datastore/BaseKeyTest.java

```diff
@@ -71,13 +86,29 @@ public void testBadDatasetInConstructor() throws Exception {
   @Test(expected = IllegalArgumentException.class)
   public void testBadDatasetInSetter() throws Exception {
     Builder builder = new Builder("d", "k");
+    builder.setProjectId(" ");
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testBadDatasetInSetterDeprecated() throws Exception {
+    Builder builder = new Builder("d", "k");
     builder.projectId(" ");
   }
 
   @Test
   public void testNamespace() throws Exception {
     Builder builder = new Builder("ds", "k");
     BaseKey key = builder.build();
+    assertTrue(key.getNamespace() != null);
+    assertTrue(key.getNamespace().isEmpty());
```

Comment:
This kind of comments are valuable. In fact, using truth would have been a good choice since the beginning.